### PR TITLE
[Android] Implement embedding API: getCompositingSurfaceType()

### PR DESCRIPTION
**Pull request by xzhan96**
Implement this API to get the rendering backend type of XWalkView.

BUG=XWALK-4867

### CODE REVIEW

**Comment by xzhan96** [comment]
@lincsoon @axinging @wuhengzhi please help to review.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 1]
```diff
@@ -262,6 +262,18 @@ public void onActivityStateChange(Activity activity, int newState) {
      */
     @XWalkAPI
     public static final int RELOAD_IGNORE_CACHE = 1;
+    /**
+     * SurfaceView as default compositing surface which has a bit performance advantage.
```
Comment (axinging) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: SurfaceView is the default compositing surface. It has less latency and uses less memory.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 2]
```diff
@@ -1823,4 +1835,15 @@ public void clearMatches() {
...
     }
+
+    /**
+     * Gets the compositing surface type of this XWalkView.
+     * @return SurfaceView or TextureView
+     * @since 7.0
+     */
+    @XWalkAPI
+    public String getCompositingSurfaceType() {
+        checkThreadSafety();
```
Comment (wuhengzhi) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: Please add `if (mContent == null) return;`

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetCompositingSurfaceTypeTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+/**
+ * Test suite for getCompositingSurfaceType().
+ */
+public class GetCompositingSurfaceTypeTest extends XWalkViewTestBase {
+
+    @Override
+    public void setUp() throws Exception {
+        XWalkPreferences.setValue(XWalkPreferences.ANIMATABLE_XWALK_VIEW, true);
```
Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetCompositingSurfaceTypeTest.java]: Extra spaces after this line.

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetCompositingSurfaceTypeTest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        XWalkPreferences.setValue(XWalkPreferences.ANIMATABLE_XWALK_VIEW, true);        
+        super.setUp();
+    }
+
+    @SmallTest
+    @Feature({"GetCompositingSurfaceType"})
+    public void testGetCompositingSurfaceType() throws Throwable {
+        String surfaceType = getCompositingSurfaceTypeOnUiThread();
+        assertEquals("TextureView", surfaceType);
+    }
```
Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetCompositingSurfaceTypeTest.java]: Test case could be more richer.

> XWalkPreferences.setValue(XWalkPreferences.ANIMATABLE_XWALK_VIEW, false);
surfaceType = getCompositingSurfaceTypeOnUiThread();
assertEquals("SurfaceView", surfaceType);

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java (hunk 2 above)
Comment (xzhan96) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: sorry, it's a typo

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetCompositingSurfaceTypeTest.java (hunk 4 above)
Comment (xzhan96) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetCompositingSurfaceTypeTest.java]: no... XWalkPreferences can not be set after XWalkView has been created.

Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetCompositingSurfaceTypeTest.java]: Yes. Forget it. It was explained in XWalkPreferencesInternal.java.

**Comment by xzhan96** [comment]
@lincsoon  please help to review and merge, thanks

**Pull request merged**